pkg/xds/server/components.go

```diff
@@ -160,6 +167,10 @@ func DefaultDataplaneSyncTracker(rt core_runtime.Runtime, reconciler, ingressRec
 					}
 				}
 
+				// Ingresses contains Services from all meshes, split them into several ingresses each of them in charge of only one mesh
+				dataplanes = ingress.SplitIngressesByMeshAndFlatten(dataplanes)
+				dataplanes = xds_topology.FilterDataplanesByMesh(dataplanes, dataplane.Meta.GetMesh())
```

Comment:
Is this splitting and filtering required? What if we will be checking mesh when iterating on `availableServices`?